src/controllers/contactSearch.controller.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+const Constants = require('../constants')
+const BaseController = require('./base.controller')
+const Contact = require('../models/contact.model')
+
+module.exports = class ContactSearchController extends BaseController {
+  static async doGet (request, reply, errors = undefined) {
+    try {
+      const context = {
+        pageTitle: 'Contact search - Waste Permits - GOV.UK'
+      }
+
+      let authToken
+      if (request.state[Constants.COOKIE_KEY]) {
+        authToken = request.state[Constants.COOKIE_KEY].authToken
+      }
+
+      // List the contacts
+      context.contacts = await Contact.list(authToken)
+
+      return reply
+        .view('contactSearch', context)
+        .state(Constants.COOKIE_KEY, request.state[Constants.COOKIE_KEY])
+    } catch (error) {
+      console.error(error)
+      return reply.redirect('/error')
+    }
+  }
+
+  static async doPost (request, reply) {
+    let authToken
+    if (request.state[Constants.COOKIE_KEY]) {
+      authToken = request.state[Constants.COOKIE_KEY].authToken
+    }
+
+    if (request.payload.id) {
+      const contact = await Contact.getById(authToken, request.payload.id)
```

Comment:
I get an `Unhandled rejected promise` message and a 500 error page when I submit an incomplete or invalid ID, and I think it's because we don't have a `.catch()` here to handle the promise rejection. Since this is just a temporary page, I'm not sure if we need to bother adding this, but thought I'd mention it.

Reply:
It's a minor thing to do so I'll ensure its covered.